gprofiler/profilers/perf.py

```diff
@@ -77,11 +81,20 @@ def _get_perf_cmd(self) -> List[str]:
             str(self._mmap_sizes[self._type]),
         ] + self._extra_args
 
+    def check_if_restart(self) -> None:
```

Comment:
```suggestion
    def check_if_needs_restart(self) -> None:
```
Let's add a docstring explaining the reasoning (what is "needs restart")